File: client/bcephal-client-model/Settings/ParameterEditorData.cs
using Bcephal.Models.Base;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Settings
{
    public class ParameterEditorData : EditorData<Parameter>
    {

        public ObservableCollection<Nameable> Grids { get; set; }

        public ObservableCollection<Nameable> Sequences { get; set; }

        public ObservableCollection<Nameable> Billtemplates { get; set; }

        public ObservableCollection<ParameterGroup> ParameterGroups { get; set; }
    }
}

File: server/bcephal-settings-service/src/main/java/com/moriset/bcephal/settings/domain/ParameterEditorData.java
/**
 * 
 */
package com.moriset.bcephal.settings.domain;

import java.util.ArrayList;
import java.util.List;

import com.moriset.bcephal.domain.EditorData;
import com.moriset.bcephal.domain.Nameable;
import com.moriset.bcephal.domain.parameter.Parameter;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class ParameterEditorData extends EditorData<Parameter> {

	private List<Nameable> grids;

	private List<Nameable> sequences;

	private List<Nameable> billtemplates;
	
	private List<ParameterGroup> parameterGroups;
	
	private List<Nameable> matGrids;

	public ParameterEditorData() {
		grids = new ArrayList<>(0);
		matGrids = new ArrayList<>(0);
		sequences = new ArrayList<>(0);
		billtemplates = new ArrayList<>(0);
	}

}
